### DISCUSSION TITLE
MinimalLib: How to remove colors from atoms (set all black)?

### DISCUSSION THREAD

**Discussion by heberleh** [rdkit/rdkit#4853] [Q&A] [answered]
I am using the MinimalLib and the  `mol.draw_to_canvas_with_highlights()` and couldn't find a solution to color all atoms in black.

I found this website pointing that colorPalette is not supported yet for the atomColourPalette
https://michelml.github.io/rdkit-js/examples/vanilla-javascript/#drawing-molecules-all-options

Is there any way to achieve that with canvas being the output?
Many thanks in advance!

**Comment by ptosco** [comment, marked as answer]
@heberleh You can get an all-black drawing as follows:
```
const mol = rdkit.get_mol("CC1(C(N2C(S1)C(C2=O)NC(=O)C(C3=CC=CC=C3)N)C(=O)O)C");
const atomColourPalette = {};
for (let i = 0; i < 119; ++i) {
  atomColourPalette[i] = [0, 0, 0];
}
drawOpts = {atomColourPalette, width: 200, height: 200};
svg = mol.get_svg_with_highlights(JSON.stringify(drawOpts));
```
![image](https://user-images.githubusercontent.com/5244385/148526137-a340a9f4-4413-4132-8459-3854b5e218a2.png)

**Reply by heberleh** [reply]
Worked like a charm! That's perfect, and I like the approach for the palette to be per atom and not atom-label.
Thank you!

**Reply by heberleh** [reply]
Hi @ptosco, do you know if #4871 is a bug? Thank you!

**Reply by heberleh** [reply]
Ah... never mind. I thought colors were per atom-index, and not atom-name; because the map uses integers!
Perhaps that explains the 119 that I missed.

**Reply by ptosco** [reply]
@heberleh exactly: I wanted to make sure that any possible atom in the periodic table would be painted in black.
